libsolidity/interface/CompilerStack.cpp

```diff
@@ -794,8 +783,15 @@ string CompilerStack::createOnChainMetadata(Contract const& _contract) const
 			);
 		}
 	}
-	meta["settings"]["optimizer"]["enabled"] = m_optimize;
-	meta["settings"]["optimizer"]["runs"] = m_optimizeRuns;
+
+	meta["settings"]["optimizer"]["orderLiterals"] = m_optimiserSettings->runOrderLiterals;
+	meta["settings"]["optimizer"]["peephole"] = m_optimiserSettings->runPeephole;
+	meta["settings"]["optimizer"]["deduplicate"] = m_optimiserSettings->runDeduplicate;
+	meta["settings"]["optimizer"]["cse"] = m_optimiserSettings->runCSE;
+	meta["settings"]["optimizer"]["constantOptimizer"] = m_optimiserSettings->runConstantOptimiser;
+	solAssert(m_optimiserSettings->constantOptimiserTradeoff < std::numeric_limits<Json::LargestUInt>::max(), "");
+	meta["settings"]["optimizer"]["constantOptimizerTradeoff"] = Json::Value(Json::LargestUInt(m_optimiserSettings->constantOptimiserTradeoff));
```

Comment:
Do you have a good suggestion for the name? I think `runs` is bad because it suggests to refer to the number of passes/runs of the optimiser.

`bytecodeSizeTradeoff`? `executionTimeTradeoff`? `sizeTimeBalance`?